gtk4/src/custom_filter.rs

```diff
@@ -0,0 +1,80 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::CustomFilter;
+use glib::object::IsA;
+use glib::translate::*;
+use std::ptr;
+
+pub trait CustomFilterExtManual: 'static {
+    fn new<F>(filter_func: F) -> CustomFilter
+    where
+        F: Fn(&glib::Object) -> bool;
+
+    fn set_filter_func<F>(&self, filter_func: F)
+    where
+        F: Fn(&glib::Object) -> bool;
+
+    fn unset_filter_func(&self);
+}
+
+unsafe extern "C" fn destroy_closure<F: Fn(&glib::Object) -> bool>(ptr: glib::ffi::gpointer) {
+    Box::<F>::from_raw(ptr as *mut _);
+}
+
+unsafe extern "C" fn trampoline<F: Fn(&glib::Object) -> bool>(
+    item: *mut glib::gobject_ffi::GObject,
+    f: glib::ffi::gpointer,
+) -> glib::ffi::gboolean {
+    let f: &F = &*(f as *const F);
+    f(&from_glib_borrow(item)).to_glib()
+}
+
+impl<O: IsA<CustomFilter>> CustomFilterExtManual for O {
+    #[doc(alias = "gtk_custom_filter_new")]
+    fn new<F>(filter_func: F) -> CustomFilter
```

Comment:
This should be implemented on `CustomFilter` and not the trait

Reply:
I did this, do I have to do something on Gir.toml?

Reply:
No change is needed